manifests/charts/gateways/istio-ingress/templates/deployment.yaml

```diff
@@ -278,6 +278,11 @@ spec:
             - path: "annotations"
               fieldRef:
                 fieldPath: metadata.annotations
+            - path: "cpu-limit"
+              resourceFieldRef:
+                containerName: istio-proxy
+                resource: limits.cpu
```

Comment:
does downward api support resource now?

Reply:
Found it [resourceFieldRef](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/#capabilities-of-the-downward-api)

Reply:
@hzxuzhonghu Yeah, actually this take previous art (for sidecar proxy only) which was removed sometime ago: https://github.com/istio/istio/pull/32928

Reply:
It wasn't fully *removed* it was *moved*. The recommended gateway install methods all use https://istio.io/latest/docs/setup/additional-setup/gateway/ which does injection which can do this in the same way.

I would prefer to just add the

```
          {{- if gt .EstimatedConcurrency 0 }}
            - --concurrency
            - "{{ .EstimatedConcurrency }}"
          {{- end -}}
```

bit to https://github.com/istio/istio/blob/c2d1f7aff60f2e7f3265839208ba0960c20f1091/manifests/charts/istio-control/istio-discovery/files/gateway-injection-template.yaml?

In current code the `--concurrency` logic in agent has caused SO many bugs, so if we are going to change it again I would have concerns there and need more thorough investigation (and testing)

Reply:
> I would prefer to just add the
> 
> ```
>           {{- if gt .EstimatedConcurrency 0 }}
>             - --concurrency
>             - "{{ .EstimatedConcurrency }}"
>           {{- end -}}
> ```

@howardjohn Yeah, but maybe requires some modification since there are some minor differences in concurrency logic between gateways and sidecar proxies.
I think it's easier for sidecar because by default the manifests in our repo still do not use injection templates, and deploy gateways like old days (egress and ingress).

Reply:
I think the long term plan is to "manage" gateways via injection and no longer support static YAML templates for gateways. I'd prefer the default to stay what it is today (no guessing at runtime).

Reply:
Looks like this is not a default path, so it's fine.